scripts/validate_rst_title_capitalization.py

```diff
@@ -0,0 +1,394 @@
+follow the convention. Particularly used for .rst files in the doc/source folder
+
+NOTE: Run from the root directory of pandas repository
+
+Examples:
+./scripts/validate_rst_title_capitalization.py doc/source/development/contributing.rst
+./scripts/validate_rst_title_capitalization.py doc/source/
+
+Files that cannot be validated: (code crashes when validating for some reason)
+doc/source/user_guide/io.rst
+doc/source/whatsnew/v0.17.1.rst
+
+Reference: doctree elements
+http://epydoc.sourceforge.net/docutils/public/docutils.nodes.Element-class.html
+
+"""
+
+import argparse
+import sys
+from docutils.parsers.rst import Parser
+import docutils
+from docutils import nodes
+import re
+import os
+from os import walk
+from typing import Generator, List
+
+
+class suppress_stdout_stderr:
+    '''
+    Code source:
+    https://stackoverflow.com/questions/11130156/suppress-stdout-stderr-print-from-python-functions
+
+    A context manager for doing a "deep suppression" of stdout and stderr in
+    Python, i.e. will suppress all print, even if the print originates in a
+    compiled C/Fortran sub-function.
+    This will not suppress raised exceptions, since exceptions are printed
+    to stderr just before a script exits, and after the context manager has
+    exited (at least, I think that is why it lets exceptions through).
+
+    This code is needed to suppress output from the parser method
+    because the parser method prints to stdout when encountering Sphinx
+    references, as it cannot parse those at this moment.
+
+    '''
+    def __init__(self):
+        # Open a pair of null files
+        self.null_fds = [os.open(os.devnull, os.O_RDWR) for x in range(2)]
+        # Save the actual stdout (1) and stderr (2) file descriptors.
+        self.save_fds = [os.dup(1), os.dup(2)]
+
+    def __enter__(self):
+        # Assign the null pointers to stdout and stderr.
+        os.dup2(self.null_fds[0], 1)
+        os.dup2(self.null_fds[1], 2)
+
+    def __exit__(self, *_):
+        # Re-assign the real stdout/stderr back to (1) and (2)
+        os.dup2(self.save_fds[0], 1)
+        os.dup2(self.save_fds[1], 2)
+        # Close all file descriptors
+        for fd in self.null_fds + self.save_fds:
+            os.close(fd)
+
+
+# Keynames that would not follow capitalization convention
+CAPITALIZATION_EXCEPTIONS = {
+    'pandas', 'Python', 'IPython', 'PyTables', 'Excel', 'JSON',
+    'HTML', 'SAS', 'SQL', 'BigQuery', 'STATA', 'Interval', 'PEP8',
+    'Period', 'Series', 'Index', 'DataFrame', 'C', 'Git', 'GitHub', 'NumPy',
+    'Apache', 'Arrow', 'Parquet', 'Triage', 'MultiIndex', 'NumFOCUS', 'sklearn-pandas'
+}
+
+# Lowercase representation of CAPITALIZATION_EXCEPTIONS
+CAPITALIZATION_EXCEPTIONS_LOWER = {word.lower() for word in CAPITALIZATION_EXCEPTIONS}
+
+# Dictionary of bad titles that will be printed later along with line numbers
+# Key: Document Directory, Value: Pair(Bad Title, Line Number)
+badTitleDictionary = {}
```

Comment:
In Python the convention is to use `bad_title_dictionary` instead of camel case. This applies to all variables, and function names in your code. Have a look at the PEP-8 style guide.